tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/ckernel_sfpu_trigonometry.h

```diff
@@ -80,82 +81,28 @@ inline void calculate_tangent()
 }
 
 template <bool APPROXIMATION_MODE>
-sfpi_inline vFloat sfpu_sine_maclaurin_series(vFloat val)
+static vFloat sfpu_sinpi(vFloat x);
+
+template <>
+sfpi_inline vFloat sfpu_sinpi<true>(vFloat x)
 {
-    // Good for [-pi:pi]
-    // Mclauren series = x - x^3/3! + x^5/5! - x^7/7! + x^9/9! - x^11/11!
-    vFloat tmp = val;
-    // x
-    vFloat output = tmp;
-    // x^3/3!
-    tmp = tmp*val*val;
-    output += -0.166666666*tmp;
-    // x^5/5!
-    tmp = tmp*val*val;
-    output +=  0.0083333333*tmp;
-    // x^7/7!
-    tmp = tmp*val*val;
-    output += -0.0001984126*tmp;
-
-    // x^9/9!
-    tmp = tmp*val*val;
-    output +=  0.0000027557*tmp;
-
-    // x^11/11!
-    tmp = tmp*val*val;
-    output += -0.00000002505*tmp;
-
-    if constexpr (not APPROXIMATION_MODE) {
-	// x^11/11!
-        tmp = tmp*val*val;
-        output += -0.00000002505*tmp;
-
-	// x^13/13!
-	tmp = tmp*val*val;
-	output += 1.6059043836821613e-10*(tmp);
-    }
+    vFloat xx = x * x;
 
-    // Write out output
-    return output;
+    return x * ((0x1.29cf02p+1f
+        * xx - 0x1.4954d4p+2f)
+        * xx + 0x1.92149p+1f);
```

Comment:
There are already two implementations in this PR.  This is the one for `APPROXIMATION_MODE = true`, and its error is around 1e-4.

![Screenshot 2024-10-07 at 21-37-40 Desmos Graphing Calculator](https://github.com/user-attachments/assets/de9a1987-a6e2-4148-b313-1a43846f25e5)

Reply:
I set that goal somewhat arbitrarily to match the precision of `fp16a`/`f16`, which is also enough for `fp16b`/`bf16`.